src/main/webapp/app/lecture/lecture.component.ts

```diff
@@ -1,85 +1,67 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import dayjs from 'dayjs/esm';
+import { Component, OnInit } from '@angular/core';
 import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
-import { Subject, Subscription } from 'rxjs';
 import { filter, map } from 'rxjs/operators';
-import { AccountService } from 'app/core/auth/account.service';
 import { LectureService } from './lecture.service';
 import { Lecture } from 'app/entities/lecture.model';
 import { ActivatedRoute, Router } from '@angular/router';
 import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
 import { onError } from 'app/shared/util/global.utils';
 import { AlertService } from 'app/core/util/alert.service';
-import { EventManager } from 'app/core/util/event-manager.service';
-import { faFile, faPencilAlt, faPlus, faPuzzlePiece, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { faFile, faPencilAlt, faPlus, faPuzzlePiece, faTimes, faFilter } from '@fortawesome/free-solid-svg-icons';
 import { LectureImportComponent } from 'app/lecture/lecture-import.component';
+import { Subject } from 'rxjs';
+
+export enum LectureDateFilter {
+    PAST = 'filterPast',
+    CURRENT = 'filterCurrent',
```

Comment:
Not all lectures follow weekly pattern. “Today” essentially means “current”. Since this is only a convenience feature for instructors, I'd consider it to be good enough.